src/Temporalio.Extensions.OpenTelemetry/ActivitySourceExtensions.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Temporalio.Workflows;
+
+namespace Temporalio.Extensions.OpenTelemetry
+{
+    /// <summary>
+    /// Temporal extensions to <see cref="ActivitySource" />.
+    /// </summary>
+    public static class ActivitySourceExtensions
+    {
+        /// <summary>
+        /// Create a <see cref="WorkflowActivity" /> and set it as current. Can only be called
+        /// within a workflow. WARNING: This method should not be considered stable and may change.
+        /// </summary>
+        /// <param name="source">The source to start an internal activity on if needed.</param>
+        /// <param name="name">Operation name.</param>
+        /// <param name="kind">Activity kind.</param>
+        /// <param name="tags">Tags for the workflow activity. See <c>inheritParentTags</c>.</param>
+        /// <param name="links">Activity links.</param>
+        /// <param name="evenOnReplay">If false (the default), the internal activity on the source
+        /// will only be started when the workflow isn't replaying. If true, the internal activity
+        /// will always be started.</param>
+        /// <param name="inheritParentTags">If true (the default), any tags on the parent
+        /// WorkflowActivity will be prepended to any given tags and used. If false, only given tags
+        /// are used. This differs from standard <c>ActivitySource.StartActivity</c> behavior by
+        /// intention so workflow tags can be propagated.</param>
+        /// <param name="updateActivity">If present and an internal activity is started from the
+        /// source, this is invoked with the started activity before it is immediately stopped.
+        /// </param>
+        /// <returns>The started workflow activity set as current.</returns>
+        /// <remarks>
+        /// This starts and immediately stops a <see cref="Activity" /> since workflows do not
+        /// support long-running .NET activities because workflows are interruptible/resumable. This
+        /// method should be used instead of any other call on <see cref="ActivitySource" /> inside
+        /// of workflows as it is replay safe. This will always return a value even if there is no
+        /// .NET activity created (e.g. no listener, during replay, etc). Users must call
+        /// <c>Dispose</c> on the activity to remove it from the current context.
+        /// </remarks>
+        public static WorkflowActivity StartWorkflowActivity(
```

Comment:
I Agree - I think there should be some distinction between those methods, they are a little analogous. Perhaps something akin to the application insights library, where you have TrackEvent - maybe StartTrackedWorkflowActivity?

Reply:
I agree, you probably should avoid anything that starts with `Workflow`, for me `Track` or `Trace` works for me, I'm leaning toward Trace due to matching with the usual 4 categories of observability eg. `Logs, Metrics, Traces and Dependencies.`

Reply:
While that overlapping is kinda rough I agree, this is an extension method on `ActivitySource`, not right next to the `Workflow` class. And this is only used by people that manually add spans to their workflows which we've found to be rare across our SDKs.

> It certainly confused me at first. Maybe we should call this WorkflowDiagnosticActivity or WorkflowTracingActivity? Bit of a mouthful but at least a bit more clear?

Concur. I like `WorkflowDiagnosticActivity`

> Perhaps something akin to the application insights library, where you have TrackEvent - maybe StartTrackedWorkflowActivity, or just TrackWorkflowActivity?

Concur. I never did like "Start" because it implies we are starting like an activity source starts, but really we start-then-stop immediately and just set it on context. I like `Track` as a verb over start.

> I agree, you probably should avoid anything that starts with Workflow

We need to make this clear it's only for workflows and not for other uses.

So I'm gonna change the method name to `TrackWorkflowDiagnosticActivity` and change the class name to `WorkflowDiagnosticActivity`.

Reply:
Updated